### ISSUE TITLE
Inconsistent implementation of TableImpl::equals and TableImpl::hashCode when comparing generated tables with plain SQL ones

### ISSUE THREAD

**Issue by AlexKolpa** [jOOQ/jOOQ#14875]
### Expected behavior

When constructing a table manually using `DSL.table(DSL.name(...))`, if you specify the schema in `DSL.name` as if it was a qualified part, it returns equals=true for any table where the name and schema matches, as per the custom `equals` check specified in `TableImpl`. Based on the hashCode contract, I would thus expect the hashCode to be the same as well.

### Actual behavior

hashCode uses the fully qualified name, thus resulting in a different hashCode, as it's not including the schema in there. This seems to be related to #7172. This causes issues with hash-based collections, such as checking whether a table is in a `Set`.

### Steps to reproduce the problem

A simple example using the provided `InformationSchema` classes is enough to reproduce. It seems to happen for any generated table:

```java
public class EqualityTest {
	public static void main(String[] args) {
		Table<Record> manualTables = DSL.table(DSL.name("information_schema", "tables"));
		Table<Record> tables = InformationSchema.INFORMATION_SCHEMA.TABLES;
		System.out.println("tables equal: " + manualTables.equals(tables));
		System.out.println("tables hashcode equal: " + (manualTables.hashCode() == tables.hashCode()));
	}
}
```

results in:
```
tables equal: true
tables hashcode equal: false
```

Given that these two tables are considered equal, I would expect the hashcodes to be equal too. Otherwise, these two tables should not be considered equal.

### jOOQ Version

JOOQ 3.18.1

### Database product and version

PostgreSQL 13.9 (Debian 13.9-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit

### Java Version

openjdk version "17.0.6" 2023-01-17

### OS Version

ArchLinux 6.2.8-arch1-1

**Comment by lukaseder** [comment]
Thanks for your report. I agree that this should be behaving according to the contracts. There's an issue somewhere to check this for all sorts of `QueryPart` types. We have some individual tests, but this might keep breaking every time we optimise some `equals()` implementation.

**Comment by lukaseder** [comment]
However, I won't backport a change in the `TableImpl::getQualifiedName` implementation, so let's just fix this `hashCode` contract problem.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0. Backport scheduled for 3.18.5 (https://github.com/jOOQ/jOOQ/issues/15098), 3.17.14 (https://github.com/jOOQ/jOOQ/issues/15099), and 3.16.20 (https://github.com/jOOQ/jOOQ/issues/15100)
